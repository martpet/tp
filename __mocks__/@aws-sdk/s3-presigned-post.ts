export const createPresignedPost = vi.fn().mockResolvedValue('dummyPresignedPost');
