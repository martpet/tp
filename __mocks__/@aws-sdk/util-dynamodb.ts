export const unmarshall = vi
  .fn()
  .mockName('unmarshall')
  .mockReturnValue('dummyUnmarshalledItems');
