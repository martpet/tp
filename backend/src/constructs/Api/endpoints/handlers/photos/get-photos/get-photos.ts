import {
  APIGatewayProxyHandlerV2,
  DynamoDBClient,
  DynamoDBDocumentClient,
  errorResponse,
  GetCommand,
  GetPhotosResponse,
  PathParam,
  photosTableOptions,
  StatusCodes,
} from 'lambda-layer';

const ddbClient = new DynamoDBClient({});
const ddbDocClient = DynamoDBDocumentClient.from(ddbClient);

export const handler: APIGatewayProxyHandlerV2<GetPhotosResponse> = async (event) => {
  const { fingerprint } = event.pathParameters as PathParam<'/photos', 'GET'>;

  if (!fingerprint) {
    return errorResponse('564208f603', { statusCode: StatusCodes.BAD_REQUEST });
  }

  const getCommand = new GetCommand({
    TableName: photosTableOptions.tableName,
    Key: { [photosTableOptions.partitionKey.name]: fingerprint },
  });

  const { Item } = await ddbDocClient.send(getCommand);

  if (!Item) {
    return errorResponse('565cca53ca', { statusCode: StatusCodes.NOT_FOUND });
  }

  return Item;
};
