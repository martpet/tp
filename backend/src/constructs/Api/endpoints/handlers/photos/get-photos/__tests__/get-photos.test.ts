import { DynamoDBDocumentClient, GetCommand } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import {
  itHasPathParam,
  itResolves,
  itResolvesWithError,
  itSendsAwsCommand,
} from '~/constructs/Api/utils';

import { handler } from '../get-photos';

vi.mock('~/constructs/Api/utils/errorResponse');

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = [
  { pathParameters: { fingerprint: 'dummyFingerprint' } },
] as unknown as Parameters<typeof handler>;

beforeEach(() => {
  ddbMock.reset();
  ddbMock.on(GetCommand).resolves({
    Item: { dummyPhotoProp: 'dummyPhotoValue' },
  });
});

describe('get-photos', () => {
  itHasPathParam('fingerprint', handler, args);
  itSendsAwsCommand(GetCommand, ddbMock, handler, args);
  itResolves(handler, args);

  describe('when "Item" is missing from "GetCommand" output', () => {
    beforeEach(() => {
      ddbMock.on(GetCommand).resolves({});
    });
    itResolvesWithError(handler, args);
  });
});
