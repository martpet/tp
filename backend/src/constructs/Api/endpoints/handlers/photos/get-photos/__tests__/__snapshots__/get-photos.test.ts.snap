// Vitest Snapshot v1

exports[`get-photos > resolves with a correct value 1`] = `
{
  "dummyPhotoProp": "dummyPhotoValue",
}
`;

exports[`get-photos > sends "GetCommand" from "DynamoDBDocumentClient" with correct args 1`] = `
{
  "Key": {
    "fingerprint": "dummyFingerprint",
  },
  "TableName": "TP-photos",
}
`;

exports[`get-photos > when "Item" is missing from "GetCommand" output > calls "errorResponse" with correct args 1`] = `
[
  [
    "565cca53ca",
    {
      "statusCode": 404,
    },
  ],
]
`;

exports[`get-photos > when "Item" is missing from "GetCommand" output > resolves with a correct value 1`] = `"dummyErrorResponse"`;

exports[`get-photos > when "fingerprint" path parameter is missing > calls "errorResponse" with correct args 1`] = `
[
  [
    "564208f603",
    {
      "statusCode": 400,
    },
  ],
]
`;

exports[`get-photos > when "fingerprint" path parameter is missing > resolves with a correct value 1`] = `"dummyErrorResponse"`;
